### DISCUSSION TITLE
How to add the Material Symbols/Icons

### DISCUSSION THREAD

**Discussion by MausVlot** [ant-design-blazor/ant-design-blazor#3965] [Q&A] [answered]
For my application I would really like to use the Material Symbols/Icons (using the official font of course).
How do I achieve this?

I added a `Icons` class which has a ` public static RenderFragment GetIcon(string iconName) => builder =>
    {
        builder.OpenElement(0, "span");
        builder.AddAttribute(0, "class", "material-symbols-outlined");
        builder.AddContent(0, iconName);
        builder.CloseElement();
    };`. 
This works fine for MenuItems because those have a IconTemplate renderfragment.
    
But with buttons we can only insert a string which then gets auto converted to a svg? 
Is there a way to add the material icons to the dictonary of svg's? Or mabye some other logic to convert prefixed strings to my custom function?
    
I saw that AntBlazor has a FontIcon element so there might be support for font icons be I cant figure it out...
    
    
Ps. I obviously I would prefer a solution that allows me to insert a string like `symbols-home` without needing to call a function.

**Comment by ElderJames** [comment, marked as answer]
Thans for contacting us @MausVlot , for now our icon inside components are mapping the icon name and svg string and render the svg element. So if you want to use other icons system, we need to add a fullback way that if the icon name is not in the list antd icons, then render the span with the class of other icons.

**Reply by MausVlot** [reply]
Ah okay, Thanks for the info. 
It would be really nice to have the option to use it the way I mentioned. (The provided icon library doesn't always provide what we are looking for but thanks for letting us know!)
